### ISSUE TITLE
controller control

### ISSUE THREAD

**Issue by nemanja-rakicevic** [cmbruns/pyopenvr#23]
Hi,
I was wondering if you could help me with the syntax regarding getting button states and sending a vibration signal (haptic pulse).
I've tried something like `openvr.IVRSystem().TriggerHapticPulse(2, 0, 500000)` to get a vibration but without success, I just get `AttributeError: 'IVRSystem' object has no attribute 'TriggerHapticPulse'
`
Thanks

**Comment by cmbruns-hhmi** [comment]
In pyopenvr, the method name is `triggerHapticPulse`, not `TriggerHapticPulse` (note capitalization). In general, pyopenvr uses method names to begin with a lower case letter, unlike the C++ OpenVR API, because that seems more Pythonic to me.

One way to see what methods are available in the IVRSystem class, for example, is to type
`>>> help(openvr.IVRSystem)`
at the python command prompt.

Also, just like in the regular C++ OpenVR API, you should probably be calling `openvr.VRSystem().triggerHapticPulse(...)` (i.e. without the "I") rather than `openvr.IVRSystem().triggerHapticPulse(...)`, to use the shared global singleton VRSystem() instance, rather than creating a newly constructed IVRSystem() every time.

I have actually tested the triggerHapticPulse method, but I expect it will work in pyopenvr exactly as it does when using the C++ OpenVR API.

In summary, use a lower case "t" and remove the "I", and you should get further.

**Comment by nemanja-rakicevic** [comment]
Yes, it was my oversight, sorry. The `getControllerState` works fine, but `triggerHapticPulse` doesn't seem to make the controllers vibrate.


PS - regarding the functions, since the openvr documentation is not quite clear, could you please tell me which of them are used to render the hmd images from a 3D envirnoment (e.g. a kinect sensor pointcloud) or which libraries should I use?
If you know of any code example which does this I would be immensely grateful!

**Comment by jzitelli** [comment]
`triggerHapticPulse` is working for me... your problem may be that the duration you are using is too large, it needs to fit in a `c_ushort`, i.e. < 2^16.

I think the only way to get longer pulses is to trigger short pulses back-to-back, and it seems like maybe you also need to time the calls to `triggerHapticPulse` so that the last pulse has finished at least 5 ms before you start a new one...  I find that if I use values ~3000, I can trigger a pulse on every frame, but higher values won't trigger any pulse.

This is what I'm doing on every frame: 
https://github.com/jzitelli/python-gltf-experiments/blob/bd871ab89c0764f143a6dc2e343d7a61624c69ea/OpenVRRenderer.py#L79-L88
